IdentityCore/src/cache/MSIDToken.h

```diff
@@ -23,22 +23,25 @@
 
 #import <Foundation/Foundation.h>
 
-typedef NS_ENUM(uint32_t, MSIDTokenType)
+typedef NS_ENUM(NSInteger, MSIDTokenType)
 {
-    MSIDTokenTypeAccessToken =  'acTk',
-    MSIDTokenTypeRefreshToken = 'rfTk'
+    MSIDTokenTypeAccessToken,
+    MSIDTokenTypeRefreshToken
 };
 
-@interface MSIDToken : NSObject
+@interface MSIDToken : NSObject <NSSecureCoding>
 
 @property (readonly) NSString *token;
-
 @property (readonly) NSString *idToken;
 @property (readonly) NSDate *expiresOn;
 @property (readonly) NSString *familyId;
 @property (readonly) NSDictionary *clientInfo;
 @property (readonly) NSDictionary *additionalServerInfo;
-
 @property (readonly) MSIDTokenType tokenType;
+@property (readonly) NSString *resource;
+@property (readonly) NSString *authority;
+@property (readonly) NSString *clientId;
+
```

Comment:
if we are to add like this, I suggest we also add scopes here too
@property (readonly) NSOrderedSet<NSString *> *scopes;

Reply:
added